### PR TITLE
fix 3d puck scaling issue

### PR DESCRIPTION
**Pull request by ank27**
### Summary of changes
Currently, in order to keep the 3d puck visible across all zoom levels, we use scale expression to calculate the scale according to the current zoom based on two anchor `maxZoom` and `minZoom` from coordinate bounds, the issue is if CoordinateBounds changes, the 3d puck size also change with it. 
this pr fixes this issue using a constant value for `minZoom` and `maxZoom`. Keeping the constants same as before so existing users don't have to scale the puck.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Fix an issue where 3D puck used to scale when changing coordinate bounds.</changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt [hunk 1]
```diff
@@ -148,11 +148,17 @@ internal class LocationPuckManager(
...
+   * current zoom level. MIN_ZOOM, MAX_ZOOM are used as two anchor points to calculate
+   * the scale expression.
+   */
   @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
   internal fun styleScaling(settings: LocationComponentSettings) {
     val puck = settings.locationPuck
-    val minZoom = delegateProvider.mapCameraManagerDelegate.getBounds().minZoom
-    val maxZoom = delegateProvider.mapCameraManagerDelegate.getBounds().maxZoom
+    val minZoom = MIN_ZOOM
+    val maxZoom = MAX_ZOOM
```
Comment (pengdev) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:161]: Shall we add some tests to cover this change?

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt [hunk 2]
```diff
@@ -148,11 +148,17 @@ internal class LocationPuckManager(
     locationLayerRenderer.hide()
   }
 
+  /**
+   * function to set scaling for [LocationPuck].
+   * In order to keep 3D puck size constant across all zoom levels, we interpolate the model based on
+   * current zoom level. MIN_ZOOM, MAX_ZOOM are used as two anchor points to calculate
+   * the scale expression.
+   */
```
Comment (pengdev) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:156]: please also update the doc of `LocationPuck3D#modelScale`

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt (hunk 1 above)
Comment (ank27) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:161]: default 3d scale test covers these.

File: plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt [hunk 3]
```diff
@@ -255,7 +255,7 @@ class LocationPuckManagerTest {
     locationPuckManager.styleScaling(settings)
     verify { locationLayerRenderer.styleScaling(capture(valueSlot)) }
     assertEquals(
-      "[interpolate, [exponential, 0.5], [zoom], 0.0, [literal, [1.0, 1.0, 1.0]], 0.0, [literal, [1.0, 1.0, 1.0]]]",
+      "[interpolate, [exponential, 0.5], [zoom], 0.5, [literal, [2965820.800757861, 2965820.800757861, 2965820.800757861]], 22.0, [literal, [1.0, 1.0, 1.0]]]",
```
Comment (Chaoba) [inline plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt:258]: Curious where are these big numbers come from?

**Review by Chaoba — APPROVED** [review]
One comment, others LGTM

File: plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt (hunk 3 above)
Comment (ank27) [reply plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt:258]: @Chaoba as we use constant MAX_ZOOM and MIN_ZOOM now, these values comes from the calculation of the scale expression using `2.0.pow(maxZoom - minZoom)`.

Comment (yunikkk) [reply plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt:258]: Could we also precalculate the `2.0.pow(maxZoom - minZoom)` and put to the constants? 
Suppose no need to calculate every time if vars are constant

Comment (ank27) [reply plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManagerTest.kt:258]: yeah will do that 👍

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt (hunk 2 above)
Comment (ank27) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:155]: will create another pr to update the doc.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt [hunk 4]
```diff
@@ -203,6 +205,13 @@ internal class LocationPuckManager(
       }
     }
   }
+
+  private companion object {
+    const val MIN_ZOOM = 0.50
+    const val MAX_ZOOM = 22.0
+    // model scale constant is calculated using 2.0*pow(maxZoom - minZoom) to keep 3D model scale constant across zoom.
+    const val MODEL_SCALE_CONSTANT = 2965820.800757861
```
Comment (pengdev) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:213]: could we calculate it once using the minZoom/maxZoom?

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt [hunk 5]
```diff
@@ -148,40 +148,44 @@ internal class LocationPuckManager(
     locationLayerRenderer.hide()
   }
 
+  /**
+   * function to set scaling for [LocationPuck].
```
Comment (yunikkk) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt:152]:
```suggestion
   * Function to set scaling for [LocationPuck].
```

**Review by yunikkk — APPROVED** [review]

**Pull request merged**